ios/keyboardObserver/REAKeyboardEventObserver.m

```diff
@@ -99,13 +99,21 @@ - (void)updateKeyboardFrame
 
   CGFloat keyboardHeight = [self computeKeyboardHeight:keyboardView];
   for (NSString *key in _listeners.allKeys) {
-    ((KeyboardEventListenerBlock)_listeners[key])(_state, keyboardHeight);
+    KeyboardEventListenerBlock listener = _listeners[key];
+    if (listener) {
+      listener(_state, keyboardHeight);
+    }
   }
 }
 
 - (CGFloat)computeKeyboardHeight:(UIView *)keyboardView
 {
-  CGFloat keyboardFrameY = [keyboardView.layer presentationLayer].frame.origin.y;
+  CALayer *presentationLayer = [keyboardView.layer presentationLayer];
+  BOOL hasSize = presentationLayer.frame.size.width && presentationLayer.frame.size.height;
+  if (!hasSize) {
+    return 0;
```

Comment:
Just wondering if it's safe to return 0 here.

Reply:
I believe the `hasSize` checko should be moved to around [line 42](https://github.com/software-mansion/react-native-reanimated/pull/3704/files#diff-1894e21e8a2a03b928b81d9663210ce7061585dcc87a2c2712cd49b0c4d7c018R42)

I had this false positive when this function returned `0`, resulting in content being stuck behind the keyboard. After investigating, it turned out the views stack had multiple `UIInputSetHostView` instances, and only one matched the `hasSize` condition. The current code picks up the first `UIInputSetHostView`. However, it should pick the first one that has width and height. Otherwise, it just puts the first `UIInputSetHostView` into `_keyboardView`, and if it is irrelevant and has a zero size, the UI is stuck with `height` = `0`, and the only way to fix it is to re-mount the native view and hope this algo picks the correct view next time. `hasSize` here cures symptoms, but not the problem.